### DISCUSSION TITLE
q-class blocks markdown processor for footnotes

### DISCUSSION THREAD

**Discussion by chrisdaaz** [thegetty/quire#441] [Get Help] [answered]
Hi all, 

I'm having trouble using footnotes within `{{< q-class >}}` elements. It seems that the inner contents are passed as string literals in the HTML output, rather than Markdown, so the footnotes are not rendered as HTML properly. 

Here's an example:

```
{{< q-class "box" >}}

### Tax Cuts and Jobs Act 

- President Trump signed the Tax Cuts and Job Act (TCJA) into law on December 22, 2017. It took effect on January 1, 2018 and will impact tax filings until 2025.[^10]
```

Renders as:

![image](https://user-images.githubusercontent.com/24395592/184227020-89bffa32-ff24-40f6-9c37-2fdb3888be87.png)

and:

![image](https://user-images.githubusercontent.com/24395592/184227150-bbfa2af5-6b56-4315-8b62-db926d7326fd.png)

Removing the {{< q-class "box" >}} wrapper from the section fixes the problem. The `q-class` shortcode looks correct to me:

```
<div class="{{ with .Get 0 }}{{ . }}{{ end }}">
  {{ .Inner | markdownify }}
</div>
```

Any ideas?

**Comment by geealbers** [comment, marked as answer]
Hi @chrisdaaz, yes, I've had problems using the `q-class` shortcode when there's also a `q-figure` shortcode within it, I think for similar reasons involving how the inner content is passed through and the order things are processed. I've started using just straight HTML `<div>` tags in my Markdown files to get around this (and it looks like this is what we will advocate for the forthcoming Eleventy version of Quire as well). Would something like this work in your case as well?:

```
<div class="box">

### Tax Cuts and Jobs Act 

- President Trump signed the Tax Cuts and Job Act (TCJA) into law on December 22, 2017. It took effect on January 1, 2018 and will impact tax filings until 2025.[^10]

</div>
```

**Reply by chrisdaaz** [reply]
That worked, thanks!
